pkg/plugin/healthz.go

```diff
@@ -37,7 +37,12 @@ type HealthZ struct {
 func (h *HealthZ) Serve() {
 	serveMux := http.NewServeMux()
 	serveMux.HandleFunc(h.HealthCheckURL.EscapedPath(), h.ServeHTTP)
-	if err := http.ListenAndServe(h.HealthCheckURL.Host, serveMux); err != nil && err != http.ErrServerClosed {
+	server := &http.Server{
+		Addr:              h.HealthCheckURL.Host,
+		ReadHeaderTimeout: 5 * time.Second,
```

Comment:
`G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)`